src/app/features/location/map/map.component.ts

```diff
@@ -163,10 +163,8 @@ export class MapComponent implements AfterViewInit {
     this.map.on("zoomend", () => {
       const allMarkers = this.markerClusterGroup.getLayers() as L.Marker[];
       allMarkers.forEach((marker: any) => {
-        const entity = marker["entity"];
-        const highlighted = marker["highlighted"];
-        if (entity) {
-          this.addMarkerStyle(marker, entity, highlighted);
+        if (marker["hasDivIcon"]) {
```

Comment:
does that mean an object without color will not get any pin?

I think it may be more intuitive behavior if we set a default (grey?) color then and do not have a special "hasDivIcon" condition at all. Unless I am missing some edge case or aspect here?